modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -1474,6 +1476,33 @@ static void iocshLoadCallFunc(const iocshArgBuf *args)
     iocshSetError(iocshLoad(args[0].sval, args[1].sval));
 }
 
+/* set*/
+static const iocshArg iocshSetArg0 = { "name",iocshArgString};
+static const iocshArg iocshSetArg1 = { "value", iocshArgString};
+static const iocshArg *iocshSetArgs[2] = {&iocshSetArg0, &iocshSetArg1};
+static const iocshFuncDef iocshSetFuncDef = {"set",2,iocshSetArgs,
+                                              "Takes name and value\n"
+                                              "Sets ioc shell variable to value\n"
+                                              "If no value is given, unsets the variable\n"
+                                              "set without parameters reports all ioc shell variables\n"};
+static void iocshSetCallFunc(const iocshArgBuf *args)
+{
+    iocshContext *context;
+
+    if (iocshContextId) {
+        context = (iocshContext *) epicsThreadPrivateGet(iocshContextId);
+
+        if (context != NULL) {
+            if(args[0].sval !=0){
+                macPutValue(context->handle, args[0].sval, args[1].sval);
+            }
+        else {
+            macReportMacros(context->handle);
+        }
```

Comment:
This `else` block needs to be indented by 4 more spaces.

However I don't think we should expose the output from `macReportMacros()` to our users, I'd rather you just do something like this for now:
```C
    else {
        printf("    'set' with no parameters is still being implemented\n");
    }
```